test/suites/greenboot.robot

```diff
@@ -0,0 +1,123 @@
+    Microshift Disrupt Service
+    Run Keyword And Expect Error    	0 != 1
+    ...    Microshift Cleanup And Start
+    Microshift Restore Service
+    Microshift Cleanup And Start
+
+GreenbootHealth Microshift Pod Failure
+    [Documentation]    Microshift Pod Network failure
+    Microshift Disrupt Pod Network
+    Run Keyword And Expect Error    	0 != 1
+    ...    Greenboot Healthy
+    Microshift Cleanup And Start
+
+
+
+*** Keywords ***
+Setup Suite
+    [Documentation]    Set up all of the tests in this suite
+    Check Required Env Variables
+    Login MicroShift Host
+    Setup Kubeconfig
+    Update Greenboot Config
+
+Update Greenboot Config
+    [Documentation]    Upload a new configuration file to the MicroShift host
+    Upload String To File   ${GREENBOOT_CONFIG_CONTENT}    ${GREENBOOT_CONFIG_FILE}
+
+Upload User Config
+    [Documentation]    Upload a new configuration file to the MicroShift host
+    Put File    ${LOCAL_CONFIG_FILE}    /tmp/busybox_running_check.sh
+    ${stdout}    ${rc}=    Execute Command
+    ...    [ -f ${REMOTE_CONFIG_FILE} ] || sudo ln -s /tmp/busybox_running_check.sh ${REMOTE_CONFIG_FILE} 
+    ...    sudo=True    return_rc=True
+    Should Be Equal As Integers    0    ${rc}
+
+    Put Directory    ${MANIFEST_SOURCE_DIR}     /tmp/manifests
+    ${stdout}    ${rc}=    Execute Command
+    ...    cp -f /tmp/manifests/* /etc/microshift/manifests/ && sudo rm -rf /tmp/manifests
+    ...    sudo=True    return_rc=True
+    Should Be Equal As Integers    0    ${rc}
+
+Cleanup User Config
+    ${stdout}    ${rc}=    Execute Command
+    ...    rm -rf ${REMOTE_CONFIG_FILE} ; sudo rm -rf /etc/microshift/manifests/*
+    ...    sudo=True    return_rc=True
+    Should Be Equal As Integers    0    ${rc}
+
+Greenboot Healthy
+    [Documentation]    check
+    Greenboot Restart
+    Wait Until Greenboot Health Check Exited
+
+Microshift Disrupt Service
+    ${stdout}    ${rc}=    Execute Command
+    ...    dnf remove -y hostname
+    ...    sudo=True    return_rc=True
+    Should Be Equal As Integers    0    ${rc}
+    
+Microshift Restore Service
+    ${stdout}    ${rc}=    Execute Command
+    ...    dnf install -y hostname
+    ...    sudo=True    return_rc=True
+    Should Be Equal As Integers    0    ${rc}
+
+Microshift Disrupt Pod Network
+    ${configuration}=    Catenate    SEPARATOR=\n
+    ...    network:
+    ...    \ serviceNetwork:
+    ...    \ - 10.66.0.0/16
+    ...
+    Upload String To File    ${configuration}    ${MICROSHIFT_CONFIG}
+
+Microshift Restore Pod Network
+    ${stdout}    ${rc}=    Execute Command
+    ...    rm -rf ${MICROSHIFT_CONFIG}
+    ...    sudo=True    return_rc=True
+    Should Be Equal As Integers    0    ${rc}
+
+Greenboot Restart
+    Systemctl    restart  greenboot-healthcheck.service
+
+Microshift Cleanup And Start
+    Cleanup MicroShift and Keep Images
+    Systemctl    enable    --now microshift
+    Greenboot Healthy
+
+Cleanup MicroShift and Keep Images
```

Comment:
This is a duplication of an existing [Cleanup MicroShift](https://github.com/openshift/microshift/blob/main/test/resources/microshift-process.resource#L75) function. 

Can we add an optional argument to the existing function instead of writing a new local one?